src/o365/spo/commands/file/file-move.ts

```diff
@@ -0,0 +1,462 @@
+        const allowSchemaMismatch: boolean = args.options.allowSchemaMismatch || false;
+        const requestUrl: string = this.urlCombine(webUrl, '/_api/site/CreateCopyJobs');
+        const requestOptions: any = {
+          url: requestUrl,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            'accept': 'application/json;odata=nometadata'
+          }),
+          body: {
+            exportObjectUris: [sourceAbsoluteUrl],
+            destinationUri: this.urlCombine(tenantUrl, args.options.targetUrl),
+            options: { 
+              "AllowSchemaMismatch": allowSchemaMismatch,
+              "IgnoreVersionHistory":true,
+              "IsMoveMode":true,
+             }
+          },
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('CreateCopyJobs request...');
+          cmd.log(requestOptions);
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((jobInfo: any): Promise<any> => {
+
+        if (this.debug) {
+          cmd.log('CreateCopyJobs response...');
+          cmd.log(jobInfo);
+        }
+
+        const jobProgressOptions: JobProgressOptions = {
+          webUrl: webUrl,
+          accessToken: siteAccessToken,
+          copyJopInfo: jobInfo.value[0],
+          progressMaxPollAttempts: 1000, // 1 sec.
+          progressPollInterval: 30 * 60, // approx. 30 mins. if interval is 1000
+          progressRetryAttempts: 5
+        }
+
+        return this.getCopyJobProgress(jobProgressOptions, cmd);
+      })
+      .then((): void => {
+        if (this.verbose) {
+          cmd.log('DONE');
+        }
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  /**
+   * Checks if a file exists on the server relative url
+   */
+  private fileExists(tenantUrl: string, webUrl: string, sourceUrl: string, siteAccessToken: string, cmd: any): request.RequestPromise {
+    const webServerRelativeUrl: string = webUrl.replace(tenantUrl, '');
+    const fileServerRelativeUrl: string = `${webServerRelativeUrl}${sourceUrl}`;
+
+    const requestUrl = `${webUrl}/_api/web/GetFileByServerRelativeUrl('${encodeURIComponent(fileServerRelativeUrl)}')/`;
+    const requestOptions: any = {
+      url: requestUrl,
+      method: 'GET',
+      headers: Utils.getRequestHeaders({
+        authorization: `Bearer ${siteAccessToken}`,
+        'accept': 'application/json;odata=nometadata'
+      }),
+      json: true
+    };
+
+    if (this.debug) {
+      cmd.log(`fileExists request...`);
+      cmd.log(requestOptions);
+    }
+
+    return request.get(requestOptions);
+  }
+
+  /**
+   * A polling function that awaits the 
+   * queued copy job to return JobStatus = 0 meaning it is done with the task.
+   */
+  private getCopyJobProgress(opts: JobProgressOptions, cmd: CommandInstance): Promise<void> {
```

Comment:
Method that doesn't return anything, shouldn't be named `get...`. We should change it to something like `waitForJobResult`

Reply:
I named that after the api endpoint url, but should have changed it since you are right, it is a void promise.

Reply:
I've changed it in this PR and will adjust in other open PRs as well. It's a private method so little impact other than clarity for the code to maintain.